data/docs/gcp-monitoring/compute-engine/tracing.mdx

```diff
@@ -0,0 +1,412 @@
+const { OTLPTraceExporter } = require('@opentelemetry/exporter-trace-otlp-http')
+const { Resource } = require('@opentelemetry/resources')
+const { SemanticResourceAttributes } = require('@opentelemetry/semantic-conventions')
+
+// do not set headers in exporterOptions, the OTel spec recommends setting headers through ENV variables
+// https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/exporter.md#specifying-headers-via-environment-variables
+
+// highlight-start
+const exporterOptions = {
+  url: 'https://ingest.{region}.signoz.cloud:443/v1/traces',
+}
+// highlight-end
+
+const traceExporter = new OTLPTraceExporter(exporterOptions)
+const sdk = new opentelemetry.NodeSDK({
+  traceExporter,
+  instrumentations: [getNodeAutoInstrumentations()],
+  resource: new Resource({
+    // highlight-next-line
+    [SemanticResourceAttributes.SERVICE_NAME]: 'computeengine_nodejs_app',
+  }),
+})
+
+// initialize the SDK and register with the OpenTelemetry API
+// this enables the API to record telemetry
+sdk.start()
+
+// gracefully shut down the SDK on process exit
+process.on('SIGTERM', () => {
+  sdk
+    .shutdown()
+    .then(() => console.log('Tracing terminated'))
+    .catch((error) => console.log('Error terminating tracing', error))
+    .finally(() => process.exit(0))
+})
+```
+
+In this example, we have used `computeengine_nodejs_app` as the service name. You can feel free to change it appropriately as per your use-case.
+
+Depending on the choice of your region for SigNoz cloud, the ingest endpoint will vary according to this table.
+
+| Region | Endpoint |
+| --- | --- |
+| US |	ingest.us.signoz.cloud:443 |
+| IN |	ingest.in.signoz.cloud:443 |
+| EU | ingest.eu.signoz.cloud:443 |
+
+
+Step 7: Run the NodeJS applicaiton using the following command:
+
+```
+OTEL_EXPORTER_OTLP_HEADERS="signoz-access-token=<SIGNOZ_INGESTION_KEY>" node -r ./tracing.js app.js
+```
+
+`SIGNOZ_INGESTION_KEY` is the API token provided by SigNoz. You can find your ingestion key from SigNoz cloud account details sent on your email.
+
+The application is now up and running on port 8080.
+
+### Open up the port on Compute Engine instance
+
+In order to invoke the application, we need to open the port 8080 on the Compute Engine instance.
+
+- On the GCP console, navigate to the Compute Engine page. Click on the instance where the NodeJS application is running.
+- Scroll down to the `Network interfaces` section, and click on the network interface name.
+- From the left side navigation menu, click on `Firewall`. On the Firewall Policies page, click on **CREATE FIREWALL RULE** button at the top.
+- Put an appropraite name for the firewall rule, say `port-8080`, and an appropriate description.
+- In the `Targets` dropdown, you can select `All instances in the network`, and in the `Source IPv4 ranges` textbox, put `0.0.0.0/0`. This will allow ht eincoming requests from any host.
```

Comment:
```suggestion
- In the `Targets` dropdown, you can select `All instances in the network`, and in the `Source IPv4 ranges` textbox, put `0.0.0.0/0`. This will allow the incoming requests from any host.
```